src/utils/config_utils/tests/config_utils_ut.cpp

```diff
@@ -373,15 +374,405 @@ static const char* invalidConfigContentDownloadTimeout =
+    {
+        settings->keepAliveInSeconds = DEFAULT_KEEP_ALIVE_IN_SECONDS;
+    }
+
+    if (!ADUC_AgentInfo_ConnectionData_GetStringField(agent_info, "dps.clientId", &settings->clientId))
+    {
+        settings->clientId = NULL;
+    }
+
+    if (!ADUC_AgentInfo_ConnectionData_GetStringField(agent_info, "dps.userName", &settings->username))
+    {
+        settings->username = NULL;
+    }
+
+    if (!ADUC_AgentInfo_ConnectionData_GetStringField(agent_info, "dps.password", &settings->password))
+    {
+        settings->password = NULL;
+    }
+
+    if (!ADUC_AgentInfo_ConnectionData_GetStringField(agent_info, "dps.caFile", &settings->caFile))
+    {
+        settings->caFile = NULL;
+    }
+
+    if (!ADUC_AgentInfo_ConnectionData_GetStringField(agent_info, "dps.certFile", &settings->certFile))
+    {
+        settings->certFile = NULL;
+    }
+
+    if (!ADUC_AgentInfo_ConnectionData_GetStringField(agent_info, "dps.keyFile", &settings->keyFile))
+    {
+        settings->keyFile = NULL;
+    }
+
+    if (!ADUC_AgentInfo_ConnectionData_GetStringField(agent_info, "dps.keyFilePassword", &settings->keyFilePassword))
+    {
+        settings->keyFilePassword = NULL;
+    }
+
+    result = true;
+
+done:
+    ADUC_ConfigInfo_ReleaseInstance(config);
+    if (!result)
+    {
+        _free_mock_adps_settings(settings);
+    }
+    return result;
+}
+
+/**
+ * @brief Struct containing the connection settings for the MQTT client.
+ * This is a data structure that is used for unit testing only.
+ * The actual MQTT broker connection settings are defined in src/extensions/agent_modules/communication_modules/inc.
+ * This data structure is used to compare the values read from the config file with the expected values.
+ */
+typedef struct _mock_mqtt_broker_settings
+{
+  char* caFile;            /*< Path to a PEM file with the chain required to trust the TLS endpoint certificate */
+  char* certFile;          /*< Path to a PEM file to establish X509 client authentication */
+  char* clientId;          /*< MQTT Client Id */
+  char* hostname;          /*< FQDM to the MQTT Broker endpoint, eg: mybroker.mydomain.com */
+  char* keyFile;           /*< Path to a KEY file to establish X509 client authentication */
+  char* keyFilePassword;   /*< Password (aka pass-phrase) to protect the key file */
+  char* password;          /*< MQTT Password to authenticate the connection */
+  char* username;          /*< MQTT Username to authenticate the connection */
+  unsigned int keepAliveInSeconds; /*< Seconds to send the ping to keep the connection open */
+  unsigned int tcpPort;    /*< TCP port to access the endpoint eg: 8883 */
+  bool cleanSession;       /*< MQTT Clean Session, might require to set the ClientId [existing sessions are not supported now] */
+  bool useTLS;             /*< Disable TLS negotiation (not recommended for production)*/
+} Mock_MQTT_Broker_Settings;
+
+static void _free_mock_mqtt_broker_settings(Mock_MQTT_Broker_Settings* settings)
```

Comment:
nit: let's not prepend with underscore `_` that is reserved for compiler internals although it's pedantic since this symbol will not clash (maybe `s_` prefix for static)?

same for line 677

Reply:
Fixed